model-optimizer/unit_tests/extensions/ops/sparse_reshape_test.py

```diff
@@ -6,45 +6,112 @@
+                                            input_actual_shape=[1, 30],
+                                            new_shape=[1, dyn],
+                                            ref_out_shape=[1, 30])
 
-        # get the result
-        res_output_indices_shape = graph.node['output_indices']['shape']
-        res_output_shape_value = graph.node['output_shape']['value']
+    def test_sparse_shape_5(self):
+        # ref_output_indices_shape = np.array([11, 3], dtype=np.int32)
+        self.build_and_test_shape_inference(input_indices_sparse_shape=[11, 2],
+                                            input_actual_shape=[dyn, 30],
+                                            new_shape=[1, dyn],
+                                            ref_out_shape=[1, dyn])
 
-        self.assertTrue(np.array_equal(ref_output_indices_shape, res_output_indices_shape),
-                        'shapes do not match expected: {} and given: {}'.format(ref_output_indices_shape, res_output_indices_shape))
-        self.assertTrue(np.array_equal(ref_output_shape_value, res_output_shape_value),
-                        'values do not match expected: {} and given: {}'.format(ref_output_shape_value, res_output_shape_value))
+    def test_sparse_shape_6(self):
+        # ref_output_indices_shape = np.array([11, 3], dtype=np.int32)
+        sparse_shape = [11, 2]
+        input_indices_value = np.arange(0, np.prod(sparse_shape)).reshape(sparse_shape)
+        self.build_and_test_shape_inference(input_indices_sparse_shape=sparse_shape,
+                                            input_actual_shape=[1, 30],
+                                            new_shape=[1, dyn],
+                                            ref_out_shape=[1, 30],
+                                            input_indices=input_indices_value,
+                                            ref_out_indices=input_indices_value)
 
+    def test_sparse_shape_7(self):
+        # ref_output_indices_shape = np.array([11, 3], dtype=np.int32)
+        sparse_shape = [11, 2]
+        self.build_and_test_shape_inference(input_indices_sparse_shape=sparse_shape,
+                                            input_actual_shape=[1, 30],
+                                            new_shape=[1, 15, 2],
+                                            ref_out_shape=[1, 15, 2])
+    # negative test with uncompatible shapes
+    def test_sparse_shape_8(self):
+        # ref_output_indices_shape = np.array([11, 3], dtype=np.int32)
+        sparse_shape = [11, 2]
+        with self.assertRaisesRegex(Error, 'Stopped shape/value propagation'):
+            self.build_and_test_shape_inference(input_indices_sparse_shape=sparse_shape,
+                                                input_actual_shape=[1, 30],
+                                                new_shape=[1, 64, 2],
+                                                ref_out_shape=[1, 15, 2])
```

Comment:
please add test cases with several `-1` and/or `dyn` in `new_shape`

Reply:
done